demos/pinter/src/main.rs

```diff
@@ -0,0 +1,116 @@
+extern crate image;
+extern crate dungenon;
+extern crate cast;
+extern crate rand;
+
+use std::io;
+use std::path::PathBuf;
+
+use dungenon::level::GridLevel as Level;
+use dungenon::tile::{Faction};
+use dungenon::generator::{FactionGen};
+use dungenon::util::Error;
+
+use rand::os::OsRng;
+use rand::XorShiftRng;
+use rand::Rand;
+use rand::distributions::{IndependentSample, Range};
+
+use image::RgbImage;
+use image::Rgb;
+
+use std::collections::HashMap;
+
+fn main() {
+    let mut colors = HashMap::new();
+    let mut index = 0;
+    for r in 0..256 {
+        for g in 0..256 {
+            for b in 0..256 {
+                colors.insert(Faction::Faction(index), Rgb([r as u8, g as u8, b as u8]));
+                index += 1;
+            }
+        }
+    }
+    println!("Initializing level...");
+    println!("Input level width: ");
+    let x = usize_from_cmd();
+
+
+    println!("Input level height: ");
+    let y = usize_from_cmd();
+
+    let mut level = Level::new_filled_with(Faction::Neutral, x, y);
+    populate_level(&mut level, index);
+    let mut factiongen = FactionGen::new();
+    println!("How many iterations");
+    let iterations = u64_from_cmd();
+    let mut buffer = level.clone();
+    for i in 0..iterations {
+        factiongen.generate(&mut level, &mut buffer);
+        println!("{} iterations done.", i);
+    }
+    faction_png_export(String::from("picture"), &colors, &mut level);
+}
+
+fn populate_level(level: &mut Level<Faction>, rangemax: usize) {
+    let range = Range::new(0,rangemax);
+    let mut rand = XorShiftRng::rand(&mut OsRng::new().unwrap());
```

Comment:
This means you cannot replay the same generation... Maybe take seed from user.

Reply:
this we can do later.